lib/easypost/services/shipment.rb

```diff
@@ -11,38 +11,62 @@ def create(params = {}, with_carbon_offset = false)
       shipment: params,
       carbon_offset: with_carbon_offset,
     }
+    response = @client.make_request(:post, 'shipments', wrapped_params)
 
-    @client.make_request(:post, 'shipments', MODEL_CLASS, wrapped_params)
+    EasyPost::InternalUtilities::Json.convert_json_to_object(response, MODEL_CLASS)
   end
 
   # Retrieve a Shipment.
   def retrieve(id)
-    @client.make_request(:get, "shipments/#{id}", MODEL_CLASS)
+    response = @client.make_request(:get, "shipments/#{id}")
+
+    EasyPost::InternalUtilities::Json.convert_json_to_object(response, MODEL_CLASS)
   end
 
   # Retrieve a list of Shipments
   def all(params = {})
-    response = @client.make_request(:get, 'shipments', MODEL_CLASS, params)
+    filters = {
+      key: 'shipments',
+      purchased: params[:purchased],
+      include_children: params[:include_children],
+    }
+
+    response = get_all_helper('shipments', MODEL_CLASS, params, filters)
     response.define_singleton_method(:purchased) { params[:purchased] }
     response.define_singleton_method(:include_children) { params[:include_children] }
+
     response
   end
 
   # Get the next page of shipments.
   def get_next_page(collection, page_size = nil)
-    get_next_page_helper(collection, collection.shipments, 'shipments', MODEL_CLASS, page_size)
+    raise EasyPost::Errors::EndOfPaginationError.new unless more_pages?(collection)
```

Comment:
Why not just pass along the filters hash directly, rather than extracting specific parameters into a separate hash? If the base method did this, wouldn't need to re-implement this on a per-service basis.

Reply:
There's no base `get_next_page` anymore; it's handled through each service's `all` method which, in turn, calls the base `get_all_helper`.

Also `before_id` and `page_size` must be intentionally left out so as not to be carried to the `get_next page` request as that'd mess up with pagination. `page_size` must always be provided by user, whereas `before_id` depends on the current page.